daprblog/content/posts/2023/dapr_flomesh/index.md

```diff
@@ -0,0 +1,546 @@
+
+- `flomesh.sh` - provides no parameters, the script will create 4 clusters, complete the environment installation and configuration, and run the demo
+- `flomesh.sh -h` - prints help information
+- `flomesh.sh -i` - creates 4 clusters, completes environment installation and configuration
+- `flomesh.sh -d` - runs the demo
+- `flomesh.sh -r` - cleans up demo-related resources
+- `flomesh.sh -u` - deletes all clusters
+- 
+Execute the following command to complete the environment installation and configuration, and run the demo.
+
+```shell
+curl -sL https://raw.githubusercontent.com/addozhang/flomesh-dapr-demo/main/flomesh.sh | bash -
+```
+
+## Step-by-Step Demonstration
+### Prerequisites
+
+To perform the demonstration, we need the following tools:
+
+- Docker
+- Kubectl
+- K3d
+- Helm
+- kubectx
+
+
+### Creating Multiple Clusters
+
+Obtain the IP address of the local machine as the communication address between clusters.
+
+```shell
+export HOST_IP=10.0.0.13
+```
+
+To create four clusters named `control-plane`, `cluster-1`, `cluster-2`, and `cluster-3`, execute the following command:
+
+```shell
+API_PORT=6444 #6444 6445 6446 6447
+PORT=80 #81 82 83
+for CLUSTER_NAME in control-plane cluster-1 cluster-2 cluster-3
+do
+  k3d cluster create ${CLUSTER_NAME} \
+    --image docker.io/rancher/k3s:v1.23.8-k3s2 \
+    --api-port "${HOST_IP}:${API_PORT}" \
+    --port "${PORT}:80@server:0" \
+    --servers-memory 4g \
+    --k3s-arg "--disable=traefik@server:0" \
+    --network multi-clusters \
+    --timeout 120s \
+    --wait
+    ((API_PORT=API_PORT+1))
+    ((PORT=PORT+1))
+done
+```
+
+This command will create four `k3d` clusters and install and configure the necessary components for the Flomesh service mesh on each cluster.
+
+### Install FSM
+
+```shell
+helm repo add fsm https://charts.flomesh.io
+helm repo update
+export FSM_NAMESPACE=flomesh
+export FSM_VERSION=0.2.1-alpha.3
+for CLUSTER_NAME in control-plane cluster-1 cluster-2 cluster-3
+do 
+  kubectx k3d-${CLUSTER_NAME}
+  sleep 1
+  helm install --namespace ${FSM_NAMESPACE} --create-namespace --version=${FSM_VERSION} --set fsm.logLevel=5 fsm fsm/fsm
+  sleep 1
+  kubectl wait --for=condition=ready pod --all -n $FSM_NAMESPACE --timeout=120s
+done
+```
+
+To join the clusters `cluster-1`, `cluster-2`, and `cluster-3` to the `control-plane` cluster, run the following command:
+
+
+```shell
+kubectx k3d-control-plane
+sleep 1
+PORT=81
+for CLUSTER_NAME in cluster-1 cluster-2 cluster-3
+do
+  kubectl apply -f - <<EOF
+apiVersion: flomesh.io/v1alpha1
+kind: Cluster
+metadata:
+  name: ${CLUSTER_NAME}
+spec:
+  gatewayHost: ${HOST_IP}
+  gatewayPort: ${PORT}
+  kubeconfig: |+
+`k3d kubeconfig get ${CLUSTER_NAME} | sed 's|^|    |g' | sed "s|0.0.0.0|$HOST_IP|g"`
+EOF
+((PORT=PORT+1))
+done
+```
+
+### Install `osm-edge`
+
+Download [osm-edge](https://github.com/flomesh-io/osm-edge) Service Mesh CLI by running below command:
```

Comment:
I don't think you mention osm-edge before now - I think it would be good to give a little one sentence description of osm-edge earlier in the blog post, where you introduce Flomesh.